numba/tests/test_tuples.py

```diff
@@ -520,6 +520,25 @@ def foo():
         r = foo()
         self.assertEqual(r, Rect(width=10, height='somestring'))
 
+    def test_dispatcher_mistreat(self):
+        # Test for issue #5215 that mistreat namedtuple as tuples
+        @jit(nopython=True)
+        def foo(x):
+            return x
+
+        in1 = (1, 2,  3)
+        out1 = foo(in1)
+        self.assertEqual(in1, out1)
+
+        in2 = Point(1, 2, 3)
+        out2 = foo(in2)
+        self.assertEqual(in2, out2)
+
+        # Check the signatures
+        self.assertEqual(len(foo.nopython_signatures), 2)
+        self.assertEqual(foo.nopython_signatures[0].args[0], typeof(in1))
+        self.assertEqual(foo.nopython_signatures[1].args[0], typeof(in2))
+
```

Comment:
wondering if this ought to test that two named tuples with the same fields are also "different" (as per original issue)?